qcodes/instrument_drivers/oxford/kelvinox.py

```diff
@@ -0,0 +1,534 @@
+                           unit='K',
+                           get_cmd=self._get_still_power,
+                           set_cmd=self._set_still_power)
+        self.add_parameter('remote_status',
+                           get_cmd=self._get_remote_status,
+                           set_cmd=self._set_remote_status,
+                           vals=vals.Ints())
+
+        for valve in self._valve_map:
+            self.add_parameter('V%s_valve' % self._valve_map[valve],
+                               get_cmd=partial(self._get_valve_status, valve=valve),
+                               set_cmd=partial(self._set_valve_status, valve=valve))
+
+    def get_all(self):
+        """
+        Reads all implemented parameters from the instrument,
+        and updates the wrapper.
+        """
+        log.info('reading all settings from instrument')
+        self.snapshot(update=True)
+
+    # Functions
+    def _execute(self, message):
+        """
+        Write a command to the device
+
+        Args:
+            message (str) : write command for the device
+        """
+        log.info('Send the following command to the device: %s' % message)
+        self.visa_handle.write('@%s%s' % (self._number, message))
+        sleep(70e-3)  # wait for the device to be able to respond
+        result = self._read()
+        if result.find('?') >= 0:
+            print("Error: Command %s not recognized" % message)
+        else:
+            return result
+
+    def _read(self):
+        """
+        Reads the total bytes in the buffer and outputs as a string.
+
+        Returns:
+            message (str)
+        """
+        bytes_in_buffer = self.visa_handle.bytes_in_buffer
+        with(self.visa_handle.ignore_warning(visa.constants.VI_SUCCESS_MAX_CNT)):
+            mes = self.visa_handle.visalib.read(
+                self.visa_handle.session, bytes_in_buffer)
+        mes = str(mes[0].decode())
+        return mes
+
+    def identify(self):
+        """Identify the device"""
+        log.info('Identify the device')
+        return self._execute('V')
+
+    def remote(self):
+        """Set control to remote and unlocked"""
+        log.info('Set control to remote and unlocked')
+        self.remote_status(3)
+
+    def local(self):
+        """Set control to local and unlocked"""
+        log.info('Set control to local and unlocked')
+        self.remote_status(2)
+
+    def close(self):
+        """Safely close connection"""
+        log.info('Closing IPS120 connection')
+        self.local()
+        super().close()
+
+    def get_idn(self):
+        """
+        Overides the function of Instrument since IPS120 does not support '*IDN?'
+
+        This string is supposed to be a comma-separated list of vendor, model,
+        serial, and firmware, but semicolon and colon are also common
+        separators so we accept them here as well.
+
+        Returns:
+            A dict containing vendor, model, serial, and firmware.
```

Comment:
This produces a warning in the docs build 

/home/travis/build/QCoDeS/Qcodes/qcodes/instrument_drivers/oxford/kelvinox.py:docstring of qcodes.instrument_drivers.oxford.kelvinox.OxfordInstruments_Kelvinox_IGH.get_idn:1: WARNING: Inline emphasis start-string without end-string.

It's better to write it as ```  `*IDN?` ``` to escape the ```*```